inc/Classes/File.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace LanSuite;
+
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Filesystem\Path;
+/**
+ * Class File
+ * Generic class for file accessor functions
+ */
+
+class File
+{
+    private static string $_basePath='';
+    private string $_relativePath='';
+    private Filesystem $_fileSystem;
+
+    /**
+     * Constructor sets base path and initializes FS object
+     * 
+     * @param Filesystem $fs Filesystem-object for dependency injection
+     */
+    public function __construct(Filesystem $fs = new Filesystem)
+    {
+        $this->_fileSystem = $fs;
+        self::$_basePath = Path::canonicalize(ROOT_DIRECTORY);
+        if (!str_ends_with(self::$_basePath, '/')) {
+            self::$_basePath .='/';
+        }
+    }
+
+    /**
+     * Builds a file path with the relative path and checks for path traversals
+     * 
+     * @param  string $path the relative path to be accessed
+     * @return string|bool The full path or false if attempted directory traversal
+     */
+    public function getFullPath($path) :string|bool 
+    {
+
+        $path = Path::canonicalize(self::$_basePath . $this->_relativePath . $path);
+        //ensure that resulting path is still below the base path, return false if not
+        if (str_starts_with($path, self::$_basePath . $this->_relativePath)) {
+            return $path;
+        } else {
+            return false;
```

Comment:
What do you think about throwing an exception rather returning a boolean?
This would make the return value be always a string.

Alternative would be an empty string.

Reply:
It is worth to to raise the generic development question if we want to change exception handling from return values to exceptions.
I'm fine with that, just following (for now) the generic codestyle that the rest has

Reply:
I think it is better to keep a consistent return type rather than mixed return types. PHP allows this, but PHP is also moving into more and more type safe variations. Why also not make use of it?

Reply:
As said: That is a generic development question, I'm fine either way